configure.ac

```diff
@@ -904,36 +904,6 @@ NL_ENABLE_DOCS([auto],[NO])
 
 AM_CONDITIONAL(CHIP_BUILD_DOCS, [test "${nl_cv_build_docs}" = "yes"])
 
-#
-# OpenThread
-#
-
-NL_WITH_OPTIONAL_INTERNAL_PACKAGE(
```

Comment:
Since we need to determine whether or not we are using the linux device layer, I added the `--enable-openthread` option.
Maybe we can have both `--enable-openthread` and `--with-openthread`.

Reply:
We shouldn't need both.

In autoconf, --enable-foo is equivalent to --with-foo and --disable-foo is equivalent to --without-foo or --with-foo=no. However, the added benefit of the --with-foo syntax is that an argument can be provided that is the location of the final artifacts for that package. The NL_WITH_PACKAGE family of constructs provides a convenience semantics around that --with-foo syntax by also providing --with-foo-libs and --with-foo-includes variants for formalizing FOO_CPPFLAGS, FOO_LDFLAGS, and FOO_LIBS.